server/models/recipe.js

```diff
@@ -18,6 +18,15 @@ module.exports = (sequelize, DataTypes) => {
     },
     {}
   );
+
+  Recipe.afterCreate(async recipe => {
+    const updatedRecipe = await recipe.update(
+      { slug: `${recipe.slug}-${recipe.id}` },
+      { returning: true }
+    );
+    Object.assign(recipe, updatedRecipe);
```

Comment:
I think it can be done beforeCreate, and without the id..... 
A helper function could be whipped up to tack a random sequence(`length === 10`??)  of  alphanumeric ([a-z 0-9]) characters onto the end of each slug beforeCreate.